crates/rpc-types-debug/src/state_dump.rs

```diff
@@ -0,0 +1,99 @@
+use alloc::collections::btree_map::BTreeMap;
+use alloy_primitives::{Address, Bytes, StorageKey, StorageValue, B256, U256};
+use serde::{Deserialize, Serialize};
+use serde_with::{base64::Base64, serde_as};
+
+/// Represents the state of an account in the Ethereum state trie.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountState {
+    /// The balance of the account
+    pub balance: U256,
+    /// The nonce of the account
+    pub nonce: u64,
+    /// The root hash of the account
+    pub root: B256,
+    /// The code hash of the account
+    pub code_hash: B256,
+    /// The code of the account
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub code: Option<Bytes>,
+    /// A map of storage slots, indexed by storage key
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage: Option<BTreeMap<StorageKey, StorageValue>>,
+    /// Address only present in iterative (line-by-line) mode
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub address: Option<Address>,
+    /// If we don't have address, we can output the key
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "key")]
+    pub address_hash: Option<B256>,
+}
+
+/// Represents a state dump, which includes the root hash of the state trie,
+/// Note: There are some differences in JSON serialization compared to geth's output, such as:
+/// - The root field in geth doesn't have a 0x prefix, while here it does
+/// - The balance field of accounts in geth is a decimal string, while here it's a hexadecimal
+///   string
+/// - The value field of storage in geth doesn't have a 0x prefix, while here it does
+#[serde_as]
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StateDump {
+    /// The root hash of the state trie
+    pub root: B256,
+    /// A map of accounts, indexed by address
+    pub accounts: BTreeMap<Address, AccountState>,
+    /// Next can be set to represent that this dump is only partial, and Next
+    /// is where an iterator should be positioned in order to continue the dump.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde_as(as = "Option<Base64>")]
+    pub next: Option<Bytes>,
```

Comment:
can you elaborate on why this is base64?

Reply:
geth use []byte for the next field, is Go, []byte default will use base64 encoding for json

Reply:
great stuff :D